tsinfer/inference.py

```diff
@@ -317,47 +332,81 @@ def match_ancestors(
     sample_data,
     ancestor_data,
     *,
-    num_threads=0,
-    path_compression=True,
     recombination_rate=None,
-    mismatch_rate=None,
-    precision=None,
-    extended_checks=False,
-    engine=constants.C_ENGINE,
-    progress_monitor=None,
+    mismatch_ratio=None,
+    recombination=None,  # Deliberately undocumented: allows passing prob array directly
+    mismatch=None,  # Deliberately undocumented: allows passing prob array directly
+    path_compression=True,
+    # performance-related parameters below
+    num_threads=0,
+    precision=None,  # Deliberately undocumented
+    engine=constants.C_ENGINE,  # Deliberately undocumented
+    progress_monitor=None,  # Deliberately undocumented
+    extended_checks=False,  # Deliberately undocumented
 ):
     """
-    match_ancestors(sample_data, ancestor_data, *, num_threads=0, path_compression=True)
+    match_ancestors(sample_data, ancestor_data, *, recombination_rate=None,\
+        mismatch_ratio=None, path_compression=True, num_threads=0)
 
-    Runs the ancestor matching :ref:`algorithm <sec_inference_match_ancestors>`
+    Run the ancestor matching :ref:`algorithm <sec_inference_match_ancestors>`
     on the specified :class:`SampleData` and :class:`AncestorData` instances,
     returning the resulting :class:`tskit.TreeSequence` representing the
     complete ancestry of the putative ancestors.
 
+    The ``mismatch_ratio`` parameter determines the balance of recombination to
+    multiple mutations at a site. Setting a high ``mismatch_ratio`` will result in
+    tree sequences with more recurrent mutations and fewer recombinations (and edges).
+    Setting a low value will result in tree sequences with more recombination events
+    and edges, and fewer mutations. In the limit, as the mismatch_ratio tends to zero,
+    only one mutation will be inferred per variable site. This is the default
+    behaviour if no ``recombination_rate`` is given. Alternatively, if
+    ``recombination_rate`` is set, ``mismatch_ratio`` defaults to a setting that
+    has been shown to give reasonable results in simulated inference.
+
+    .. note::
+        As a rough guide, mismatch ratios between 1e-3 and 1e3 have been found to
+        give reasonable results for simulated, error-prone human-like data.
+
     :param SampleData sample_data: The :class:`SampleData` instance
         representing the input data.
     :param AncestorData ancestor_data: The :class:`AncestorData` instance
-        representing the set of ancestral haplotypes that we are finding
-        a history for.
-    :param int num_threads: The number of match worker threads to use. If
-        this is <= 0 then a simpler sequential algorithm is used (default).
+        representing the set of ancestral haplotypes for which we are finding
+        a history.
+    :param recombination_rate: Either a RateMap object, or a floating
+        point value giving a single rate, :math:`\\rho`, across the entire sequence.
```

Comment:
I think we should set aside a section of the documentation to describe and discuss the recombination and mismatch parameters in detail rather than trying to do it here in the context of the API docs. E.g. "See :ref:`sec_model_recombination_mutation`` for details on this parameter and how it interacts with the mismatch process".

We don't need to write this section now, can just make it a TODO for the moment (probably better, in order to get this PR merged ASAP).